### PR TITLE
Update interview actions to only be visible to users with correct permissions

### PR DESCRIPTION
**Pull request by despo**
## Context

Only users with Set up interview permissions should be able to access the Create, Cancel and Edit buttons on the interviews page

## Changes proposed in this pull request

Amendments to the authorisation service to take into account both the training and accredited provider user permissions, as well as the application visiblity,

## Guidance to review

- Enable the interview feature flag
- Assign Make decisions permissions only to your user
- Create a new interview (the actions on the application choice page will be amended as part of a different PR)
- You should not be able to perform any further actions on the /interviews page
- Grant your user Set up interview permissions
- You should not be able to Create, Cancel and Edit interviews.

## Link to Trello card

https://trello.com/c/2KjUSGHv/3919-update-interview-actions-to-only-be-visible-to-users-with-correct-permissions

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/services/provider_authorisation.rb [hunk 1]
```diff
@@ -74,14 +74,21 @@ def can_manage_organisation?(provider:)
     @actor.provider_permissions.exists?(provider: provider, manage_organisations: true)
   end
 
-  def can_set_up_interviews?(provider:)
+  def can_set_up_interviews?(application_choice:, course_option:)
```
Comment (steventux) [inline app/services/provider_authorisation.rb:77]: :+1: Makes sense to add these finer grained checks.

**Review by steventux — APPROVED** [review]
Looks good, I wonder if there's additional controller security needed for the scenario where the feature flag is active and we're allowing access to CRUD actions.

File: app/controllers/provider_interface/interviews_controller.rb [hunk 2]
```diff
@@ -9,11 +9,17 @@ def index
...
       )
-      @provider_can_make_decisions = current_provider_user.authorisation.can_make_decisions?(
-        application_choice: @application_choice,
-        course_option_id: @application_choice.current_course_option.id,
-      )
-      @interviews_can_be_created_and_edited = application_at_interviewable_stage && @provider_can_make_decisions
+      @provider_can_make_decisions =
+        current_provider_user.authorisation.can_make_decisions?(application_choice: @application_choice,
+                                                                course_option: @application_choice.current_course_option)
+      if FeatureFlag.active?(:interview_permissions)
```
[the remark this answers is not in the document]
Comment (despo) [reply app/controllers/provider_interface/interviews_controller.rb:15]: Yes, I completely missed that!

[the remark this answers is not in the document]
Comment (despo) [reply app/controllers/provider_interface/interviews_controller.rb:16]: Done. It was getting quite complex attempting to do partial checks for `Make decisions` though so we can refine as part of the next card. I'll make a note there.

**Review by steventux — APPROVED** [review]

**Pull request merged**